src/aleph/handlers/content/vm.py

```diff
@@ -320,15 +320,19 @@ class VmMessageHandler(ContentHandler):
     """
 
     async def check_balance(self, session: DbSession, message: MessageDb) -> None:
-        if message.type != MessageType.instance:
+        if not (message.type == MessageType.instance) and not (message.type == MessageType.program):
             return
+
         content = _get_vm_content(message)
-        if isinstance(content, ProgramContent):
-            return
+
+        if message.type == MessageType.program:
```

Comment:
It's easier for mypy to figure out what's going on if you use `if isinstance(content, ProgramContent)`.